src/plugins/conditionals/conditionals.c

```diff
@@ -218,14 +218,26 @@ static CondResult evalCondition (const char * leftSide, Comparator cmpOp, const
 		else if (rightSide && elektraStrLen (rightSide) > 1)
 		{
 			// not a literal, it has to be a key
-			len = keyGetNameSize (parentKey) + elektraStrLen (rightSide);
+			if (rightSide[0] == '@')
+				len = keyGetNameSize (parentKey) + elektraStrLen (rightSide);
+			else if (!strncmp (rightSide, "..", 2) || !strncmp (rightSide, ".", 1))
+				len = keyGetNameSize (curKey) + elektraStrLen (rightSide);
+			else
+				len = elektraStrLen (rightSide);
+
 			if (elektraRealloc ((void **)&lookupName, len) < 0)
 			{
 				ELEKTRA_SET_ERROR (87, parentKey, "Out of memory");
 				result = ERROR;
 				goto Cleanup;
 			}
-			snprintf (lookupName, len, "%s/%s", keyName (parentKey), rightSide);
+			if (rightSide[0] == '@')
+				snprintf (lookupName, len, "%s/%s", keyName (parentKey), rightSide + 1);
+			else if (rightSide[0] == '.')
```

Comment:
inconsistent usage, why not using `!strncmp (rightSide, ".", 1)` again?

Some lines of comments would be very useful.